src/envoy/http/authn/authn_utils.cc

```diff
@@ -22,47 +22,31 @@ namespace Http {
 namespace Istio {
 namespace AuthN {
 namespace {
-class JsonIterCallback : public Logger::Loggable<Logger::Id::filter> {
- public:
-  // The json iterator callback function
-  static bool JsonIteratorCallback(const std::string&,
-                                   const Envoy::Json::Object&,
-                                   istio::authn::JwtPayload* payload);
-};
-
-bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
-                                            const Envoy::Json::Object& obj,
-                                            istio::authn::JwtPayload* payload) {
-  ENVOY_LOG(debug, "{}: key is {}", __FUNCTION__, key);
-  ::google::protobuf::Map< ::std::string, ::std::string>* claims =
-      payload->mutable_claims();
+// Extract JWT audience into the JwtPayload.
+// Return true if "aud" is extracted. Otherwise, return false.
+bool ExtractJwtAudience(const Envoy::Json::Object& obj,
+                        istio::authn::JwtPayload* payload) {
+  const std::string key = "aud";
+  // "aud" can be either string array or string.
+  // First, try as string, will throw execption if object type is not string.
+  try {
+    std::string obj_str = obj.getString(key);
+    // Save "aud" to the payload
+    payload->add_audiences(obj_str);
+    return true;
+  } catch (Json::Exception& e) {
+    // Not convertable to string
+  }
+  // Next, try as string array
   try {
-    if (key == "aud") {
-      // "aud" can be either string array or string.
-      // Try as string array, read it as empty array if doesn't exist.
-      std::vector<std::string> aud_vector;
-      try {
-        aud_vector = obj.getStringArray(key, true);
-      } catch (Json::Exception& e) {
-        // Try as string
-        try {
-          auto audience = obj.asString();
-          aud_vector.push_back(audience);
-        } catch (Json::Exception& e) {
-          ENVOY_LOG(error, "aud field type is not string or string array");
-        }
-      }
-      for (size_t i = 0; i < aud_vector.size(); i++) {
-        payload->add_audiences(aud_vector[i]);
-      }
-    } else {
-      // will throw execption if value type is not string.
-      // In current implementation, only string objects are extracted into
-      // claims.
-      (*claims)[key] = obj.asString();
+    std::vector<std::string> aud_vector = obj.getStringArray(key);
+    for (size_t i = 0; i < aud_vector.size(); i++) {
```

Comment:
Use c11 syntax:

for (const auto& aud : aud_vector) {
   payload->add...
}

Reply:
Done.